src/executor/mod.rs

```diff
@@ -1,5 +1,7 @@
 //! Types which can take a boxed closure and execute it, preferably in the background.
 
+#![cfg_attr(feature="clippy", allow(boxed_local))]
```

Comment:
Here is the warning (see [here](https://github.com/Manishearth/rust-clippy/wiki#boxed_local) for the explanation):

```
error: local variable doesn't need to be boxed here
  --> src/executor/mod.rs:36:13
   |
36 |     fn exec(self: Box<Self>) {
   |             ^^^^
   |
   = note: #[deny(boxed_local)] implied by #[deny(clippy)]
```